test/ConstructiveSolidGeometry/CSG_test.jl

```diff
@@ -131,6 +131,26 @@ using SolidStateDetectors.ConstructiveSolidGeometry:
                     @test !(CartesianPoint{T}(0.9*sphere.r, 0, 0) in sphere)
                     @test !(CartesianPoint{T}(0, 0.1*sphere.r, sphere.r) in sphere)
                 end
+                @testset "RegularPolygon" begin
+                    pentagon = RegularPolygon(5, 0, 2, 0) # r from 0..2, z = 0
+                    @test CartesianPoint{T}(0, 0, 0) in pentagon
+                    @test CartesianPoint{T}(2, 0, 0) in pentagon
+                    @test !(CartesianPoint{T}(0, 2, 0) in pentagon)
+                    @test !(CartesianPoint{T}(0, 0, 0.1) in pentagon)
+                    @test !(CartesianPoint{T}(2.1, 0, 0) in pentagon)
+                end
+                @testset "RegularPrismMantle" begin
+                    prism = RegularPrismMantle(4, 1, 0, 1) # z from 0..1, r = 1
+                    @test !(CartesianPoint{T}(0, 0, 0) in prism)
+                    @test CartesianPoint{T}(1, 0, 0) in prism
+                    #@test CartesianPoint{T}(-1, 0, 0) in prism
```

Comment:
Note that some errors with in() occur at borders with Float32 (but not Float64) (commented lines) to be addressed along with other rounding errors

Reply:
We'll have to see if we redefine `in()` with an absolute tolerance to deal with rounding errors

Reply:
Note that these were fixed by proper typing of the PrismMantle